### PR TITLE
Fix call to nil stat

### LINKED ISSUE #10242: Panic in Volume.createIfNotExist
I'm running build from master, getting the next error:

```
time="2015-01-21T19:30:34+03:00" level="warning" msg="WARNING: Udev sync is not supported. This will lead to unexpected behavior, data loss and errors"
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x73c61c]

goroutine 9 [running]:
github.com/docker/docker/volumes.(*Volume).createIfNotExist(0xc2085071d0, 0x0, 0x0)
	/go/src/github.com/docker/docker/volumes/volume.go:91 +0xbc
github.com/docker/docker/volumes.(*Volume).initialize(0xc2085071d0, 0x0, 0x0)
	/go/src/github.com/docker/docker/volumes/volume.go:111 +0x9a
github.com/docker/docker/volumes.(*Repository).restore(0xc20850a1e0, 0x0, 0x0)
	/go/src/github.com/docker/docker/volumes/repository.go:100 +0x402
github.com/docker/docker/volumes.NewRepository(0xc208203980, 0x17, 0x7fc4d2e77c40, 0xc2081a68f0, 0x17, 0x0, 0x0)
	/go/src/github.com/docker/docker/volumes/repository.go:39 +0x1fa
github.com/docker/docker/daemon.NewDaemonFromDirectory(0x12d7c40, 0xc208078a50, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/daemon.go:892 +0x1114
github.com/docker/docker/daemon.NewDaemon(0x12d7c40, 0xc208078a50, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/daemon.go:779 +0x44
main.func·001()
	/go/src/github.com/docker/docker/docker/daemon.go:55 +0x46
created by main.mainDaemon
	/go/src/github.com/docker/docker/docker/daemon.go:78 +0x35e
```

Problematic snippet of code:

```go
	if stat, err := os.Stat(v.Path); err != nil && os.IsNotExist(err) {
		if stat.IsDir() {
			os.MkdirAll(v.Path, 0755)
		}
	// ...
	}
```

`err` is actually `stat error: stat : no such file or directory`.

### PR DESCRIPTION
**Pull request by cpuguy83**
Fixes #10242

### CODE REVIEW

**Comment by LK4D4** [comment]
Hehe

**Comment by cpuguy83** [comment]
Drone having issues pushing/pulling it seems.

**Comment by tiborvass** [comment]
@cpuguy83 but do you know why that mkdir part was there?

**Comment by cpuguy83** [comment]
Actually, I'm looking at this, it should need the MkdirAll, but not the other stuff.
It seems binds are being created here: https://github.com/docker/docker/blob/47e3da848ffbe88d0188ae6cfc09d6e1668bc293/daemon/start.go#L54-L66
Which is why we weren't getting the panic.

Removing this block of code is giving some other weird issues atm.

**Comment by crosbymichael** [comment]
Can you rebase or something to make drone happy?

**Comment by cpuguy83** [comment]
This fixes the panic only, but I will need to look at the binds setup in daemon/start.go for another PR.

**Comment by crosbymichael** [comment]
What is wrong with binds?

**Comment by cpuguy83** [comment]
Nothing is "wrong" other than having the bind paths created in "setupHostConfig" was masking the issue with the stat here, and it really should be created by the volumes system.

**Comment by crosbymichael** [comment]
Ok, SGTM

**Comment by jfrazelle** [comment]
LGTM

File: volumes/volume.go [hunk 1]
```diff
@@ -86,30 +86,14 @@ func (v *Volume) AddContainer(containerId string) {
...
-func (v *Volume) createIfNotExist() error {
-	if stat, err := os.Stat(v.Path); err != nil && os.IsNotExist(err) {
-		if stat.IsDir() {
-			os.MkdirAll(v.Path, 0755)
-		}
-
-		if err := os.MkdirAll(filepath.Dir(v.Path), 0755); err != nil {
-			return err
-		}
-		f, err := os.OpenFile(v.Path, os.O_CREATE, 0755)
```
Comment (crosbymichael) [inline volumes/volume.go]: And what about files?

Comment (cpuguy83) [inline volumes/volume.go]: Files wouldn't make it here since we wouldn't be creating files anyway.

Comment (crosbymichael) [inline volumes/volume.go]: I guess this was just a completely dead block of code

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**